### ISSUE TITLE
Zone system

### ISSUE THREAD

**Issue by karlek** [Lapsang-boys/Rex-Temporis#19]
The system currently only works for one set of regions in the first zone, it needs to work for several regions using PlayerData to select the zone to use.

Add zone finish sound effect

**Comment by karlek** [comment]
What should our zone system be able to handle?

These are a must according to me:
* Multiple spawn regions
* Multiple areas

These might be nice:
* Multiple end regions (useful when we want to make the units cross your area, i.e an X-shaped area)
* Multiple set of movement nodes (useful if we want diverging paths etc)

__Implementation__

_Spawn_

Fixing spawn is actually quite easy, we just add new spawn positions. And `getSpawn(p)` probably needs to return a list.

We might also need to move set custom value so that units can start with custom value 2 etc.

```python
// Example Area 1
SpawnPosition.Spawn1Left

// Example Area 2
SpawnPosition.Spawn1Left
SpawnPosition.Spawn2Left
```

_End_

This one is also easy, just add new rects to the `ends` list.

_Movement_

I have a proposal for solving alternate moving paths. For example look at these custom values:
```python
// Zone 2, normal route 
1
2
3
4

// Zone 2, crossing route
11
12
13
14

// Zone 2, further down route
21
22
23
24
```

Then we can subtract their first digit for which route they are taking, and look at the last digit for which node in that route they are going to.

Any thoughts @draftlord?

**Comment by godbit** [comment]
Yeah i agree with you for the most part. Im not entirely convinced about having multiple ends and so on. I think the idea about multiple spawns and maybe an difference in its route is ok. But im not sure about different ends. Of course the all depends on the layout of the zone to it is hard to say anything about the stage.

I like the thinking with customvalue. it effectively lets us add one additional information!

**Comment by godbit** [comment]
We also need to remember to think about things like:
- Rmk, this should probably rmk to the current zone start but it should also be possible to restart entirely
- Changing difficulty, should only be possible on Level 1, Zone 1

**Comment by karlek** [comment]
Agreed.

```go
// Remakes the current zone.
-rmk 
// Remakes to the specific zone (if the user has the permission)
-rmk 2
```

**Comment by karlek** [comment]
Closed with: 4845761d800ad34f61e299923cd9bd009a0bf08a
